CT-1222 Improves user journeys

aliuk2012 opened the pull request:
At the moment there is an inconsistent behaviour after a user takes an
action on a case, sometimes they are taken back to the case details page
and some times they are taken back to the case list page. In general
all actions should redirect the user back the the case details page.
After this commit the only time the user will be redirected to the case
list page is if the case has been deleted.

THe following journeys were affected

- BMT reassigns a case to another team
- Disclosure/Press/Private Office approve a cases response.
- Kilo uploads a response to a non-trigger case.
- Disclosure/Press/Private Office requests amendments to the response
  before clearing.
- Disclosure unflags a trigger case.

aliuk2012 commented on app/models/correspondence.rb:
```diff
@@ -2,6 +2,10 @@ class Correspondence < ApplicationRecord
   acts_as_gov_uk_date :received_date
 
+  scope :by_deadline, lambda {
+    order("(properties ->> 'external_deadline')::timestamp with time zone ASC")
```
No sign of sorting by ID or any other secondary sorting.

EdwardAndress commented:
@misaka are you happy with the changes I've made since your original approval?  If so, I'll squash the two commits into one, then proceed with the merge.

aliuk2012 committed:
CT-1222 Users should stay on case details page

At the moment there is an inconsistent behaviour after a user takes an
action on a case, sometimes they are taken back to the case details page
and some times they are taken back to the case list page. In general
all actions should redirect the user back the the case details page.
After this commit the only time the user will be redirected to the case
list page is if the case has been deleted.

THe following journeys were affected

- BMT reassigns a case to another team
- Disclosure/Press/Private Office approve a cases response.
- Kilo uploads a response to a non-trigger case.
- Disclosure/Press/Private Office requests amendments to the response
  before clearing.
- Disclosure unflags a trigger case.

aliuk2012 committed:
Fix page heading styling
